gvsbuild/utils/base_project.py

```diff
@@ -482,15 +480,18 @@ def add_all():
                 Project.add(c_inst, type=ty)
             else:
                 del c_inst
-        del Project._reg_prj_list
 
     def ignore(self):
         """Mark the project not to build/add to the list."""
         self.to_add = False
 
     @staticmethod
     def get_project(name):
-        return Project._dict[name]
+        try:
+            project = Project._dict[name]
+            return project
+        except KeyError:
+            log.error_exit(f"KeyError getting project {name} from Project._dict")
```

Comment:
Nit: `f"Could not find project '{name}'` (we don't want to expose the internal implementation)